### PR TITLE
Create search worker

### PR DESCRIPTION
**Pull request by jasonLaster**
Associated Issue: #3096

### Summary of Changes

* moves the `countMatches` function to a worker, which will keep the main thread from janking up when we search big files
* it also sets us up nicely to add the project text search function which *will* jank if we're not on the main thread

### CODE REVIEW

File: src/utils/pretty-print/worker.js [hunk 1]
```diff
@@ -59,20 +57,4 @@ function invertMappings(mappings: Mappings) {
...
-      response: {
-        code,
-        mappings: invertMappings(mappings)
-      }
-    });
-  } catch (e) {
-    self.postMessage({ id, error: e });
-  }
-};
+self.onmessage = workerHandler({ prettyPrint });
```
Comment (jasonLaster) [inline src/utils/pretty-print/worker.js]: did a bit of clean up while i was in the worker code

File: src/utils/parser/index.js [hunk 2]
```diff
@@ -4,20 +4,13 @@ import { workerUtils } from "devtools-utils";
...
+export const stopParserWorker = dispatcher.stop.bind(dispatcher);
 
-const getClosestExpression = dispatcher.task("getClosestExpression");
-const getSymbols = dispatcher.task("getSymbols");
-const getVariablesInScope = dispatcher.task("getVariablesInScope");
-const getOutOfScopeLocations = dispatcher.task("getOutOfScopeLocations");
+export const getClosestExpression = dispatcher.task("getClosestExpression");
+export const getSymbols = dispatcher.task("getSymbols");
+export const getVariablesInScope = dispatcher.task("getVariablesInScope");
+export const getOutOfScopeLocations = dispatcher.task("getOutOfScopeLocations");
```
Comment (jasonLaster) [inline src/utils/parser/index.js]: ooh, we an use export directly and avoid the module.exports... yay

File: src/utils/search/worker.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+): number {
+  debugger;
+  const regexQuery = buildQuery(query, modifiers, {
+    isGlobal: true
+  });
+  const match = text.match(regexQuery);
+  return match ? match.length : 0;
+}
+
+debugger;
```
Comment (nchevobbe) [inline src/utils/search/worker.js]: oops

File: src/utils/search/worker.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import buildQuery from "./utils/build-query";
+import { workerUtils } from "devtools-utils";
+const { workerHandler } = workerUtils;
+
+export function countMatches(
+  query: string,
+  text: string,
+  modifiers: SearchModifiers
+): number {
+  debugger;
```
Comment (nchevobbe) [inline src/utils/search/worker.js]: oops

**Pull request merged**